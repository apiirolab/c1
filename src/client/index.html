<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>