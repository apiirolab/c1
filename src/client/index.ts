export const info = 'Client Info';
