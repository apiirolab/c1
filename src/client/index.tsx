import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { Routes } from './config/router';
import { client } from './config/apollo';
// import * as serviceWorker from './registerServiceWorker';

ReactDOM.render(
  <ApolloProvider client={client}>
    <Routes />
  </ApolloProvider>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.unregister();
