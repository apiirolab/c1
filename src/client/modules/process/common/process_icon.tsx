import * as React from 'react';

import { List, SemanticICONS } from 'semantic-ui-react';
import styled from 'styled-components';

export function processTypeToIcon(type: string): SemanticICONS {
  switch (type) {
    case 'Travel':
      return 'briefcase';
    case 'HumanResources':
      return 'user circle';
    case 'Purchases':
      return 'money';
    case 'Sales':
      return 'money bill alternate outline';
  }
  return 'tasks';
}

const IconView = styled(List.Icon)`
  &&&&& {
    vertical-align: top !important;
    width: 25px;
  }
`;

export type Props = {
  type: string;
};

export const ProcessIcon: React.SFC<Props> = ({ type }) => (
  <IconView name={processTypeToIcon(type)} verticalAlign="middle" />
);
