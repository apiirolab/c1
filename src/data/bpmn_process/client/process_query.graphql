query ProcessQuery($id: ID!) {
  process(id: $id) {
    actionCount
    description
    id
    model
    resources {
      name
      type
    }
    status
    version
  }
}
