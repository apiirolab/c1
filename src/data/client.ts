import * as QueryTypes from './generated/types';

export { default as gql } from 'graphql-tag';

export { QueryTypes };

export function purge<T>(params: any): T {
  for (let key of Object.getOwnPropertyNames(params)) {
    if (!params[key]) {
      delete params[key];
    }
    if (params[key] instanceof Object) {
      purge(params[key]);
    }
  }
  return params;
}
