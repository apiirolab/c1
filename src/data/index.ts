export const info = 'Data Info';
