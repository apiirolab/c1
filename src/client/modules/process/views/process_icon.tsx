import * as React from 'react';

import { List, SemanticICONS } from 'semantic-ui-react';
import styled from 'styled-components';

import { QueryTypes } from 'data/client';

export function processTypeToIcon(type: QueryTypes.ProcessType): SemanticICONS {
  switch (type) {
    case QueryTypes.ProcessType.Travel:
      return 'briefcase';
    case QueryTypes.ProcessType.HumanResources:
      return 'user circle';
    case QueryTypes.ProcessType.Purchases:
      return 'money';
    case QueryTypes.ProcessType.Sales:
      return 'money bill alternate outline';
  }
  return 'tasks';
}

const IconView = styled(List.Icon)`
  &&&&& {
    vertical-align: top !important;
  }
`;

export type Props = {
  type: QueryTypes.ProcessType;
};

export const ProcessIcon: React.SFC<Props> = ({ type }) => (
  <IconView name={processTypeToIcon(type)} verticalAlign="middle" />
);
