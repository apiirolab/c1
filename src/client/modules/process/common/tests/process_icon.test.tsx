import * as React from 'react';

import { render } from 'client/tests';
import { QueryTypes } from 'data/client';
import { ProcessIcon } from '../process_icon';

describe('Process', () => {
  describe('Icon', () => {
    const Component = (
      <ul>
        <li>
          <ProcessIcon type={QueryTypes.ProcessType.HumanResources} /> Human Resources
        </li>
        <li>
          <ProcessIcon type={QueryTypes.ProcessType.Purchases} /> Purchases
        </li>
        <li>
          <ProcessIcon type={QueryTypes.ProcessType.Sales} /> Sales
        </li>
        <li>
          <ProcessIcon type={QueryTypes.ProcessType.Travel} /> Travel
        </li>
      </ul>
    );

    it('renders correctly', () => {
      expect(render(Component)).toMatchSnapshot();
    });

    return {
      component: Component
    };
  });
});
