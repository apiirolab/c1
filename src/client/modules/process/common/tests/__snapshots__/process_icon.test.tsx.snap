// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process Icon renders correctly 1`] = `
<ul>
  <li>
    <i
      aria-hidden="true"
      className="user circle icon middle aligned sc-bdVaJa jnSeCf"
    />
     Human Resources
  </li>
  <li>
    <i
      aria-hidden="true"
      className="money icon middle aligned sc-bdVaJa jnSeCf"
    />
     Purchases
  </li>
  <li>
    <i
      aria-hidden="true"
      className="money bill alternate outline icon middle aligned sc-bdVaJa jnSeCf"
    />
     Sales
  </li>
  <li>
    <i
      aria-hidden="true"
      className="briefcase icon middle aligned sc-bdVaJa jnSeCf"
    />
     Travel
  </li>
</ul>
`;
